rules/prefer-replace-all.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const quoteString = require('./utils/quote-string');
+
+function hasGlobalFlag(node) {
+	const searchPattern = node.arguments[0];
+	return searchPattern && searchPattern.regex && searchPattern.regex.flags === 'g';
+}
+
+function isLiteralCharactersOnly(node) {
+	const searchPattern = node.arguments[0].regex.pattern;
+	const specialCharacters = searchPattern.match(/[$()*+.?[\\\]^{}]/g);
+	const specialCharactersCount = specialCharacters ? specialCharacters.length : 0;
+	const escapedSpecialCharacters = searchPattern.match(/\\[$()*+.?[\\\]^{}]/g);
+	const escapedSpecialCharactersCount = escapedSpecialCharacters ? escapedSpecialCharacters.length : 0;
+	return specialCharactersCount === 2 * escapedSpecialCharactersCount;
+}
+
+function replaceNode(node, fixer) {
+	const stringName = node.callee.object.name;
+	const searchPattern = node.arguments[0].regex.pattern;
+	const replacePattern = node.arguments[1].value;
+	return fixer.replaceText(node, `${stringName}.replaceAll(${quoteString(searchPattern)}, ${quoteString(replacePattern)})`);
+}
```

Comment:
This part is not right, callee should not touched,  callee might any type not only `Identifier`,  replacement also not safe

```js
const replacement = '0';
const f00 = 'foo'.replace(/o/g, replacement);
```

I suggest `replace` -> `replaceAll`, then `arguments[0]` -> `fixed string`, `arguments[1]` should not touched